python/paddle/fluid/tests/unittests/test_program_prune_backward.py

```diff
@@ -0,0 +1,99 @@
+#   Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+
+import contextlib
+import numpy as np
+import paddle.fluid as fluid
+import paddle.fluid.core as core
+from test_parallel_executor_mnist import simple_fc_net, fc_with_batchnorm
+import seresnext_net
+from test_parallel_executor_transformer import transformer, get_feed_data_reader
+
+
+class TestProgramPruneBackward(unittest.TestCase):
+    def check_prune_correctness(self, method, feed_dict, optimizer):
+        with self.program_scope_guard():
+            loss = method(use_feed=False)
+
+            main_program = fluid.default_main_program()
+            test_prog_orig = main_program.clone(for_test=True)
+            optimizer(learning_rate=0.001).minimize(loss)
+            test_prog_prune = main_program.clone(for_test=True)
```

Comment:
Maybe we should make the model more complicated, for example, add regularization ops.

Reply:
Done